### ISSUE TITLE
Need warning for glob and single quotes in "${var:-unchecked}"

### ISSUE THREAD

**Issue by jakwings** [koalaman/shellcheck#2453]
- My shellcheck version 0.8.0-git-88cdb4e2c9b45becb21bd02cd7b205d5bef8cb56
- [x] https://www.shellcheck.net/ (i.e. the latest commit) currently gives no useful warnings about this
- [x] I searched through https://github.com/koalaman/shellcheck/issues and didn't find anything related

#### Here's a snippet or screenshot that shows the problem:

```sh
#!/bin/sh
unset -v foo
echo "${foo-"/bar"}"  #=> "/bar"
echo "${foo-'/bar'}"  #=> "'/bar'"
echo "${foo-/bar*}"   #=> "/bar*"
```

Tested using `sh/bash/zsh/dash/ksh/mksh/oksh` and `shellcheck -o all -s sh/bash/dash/ksh`.

#### Here's what shellcheck currently says:

`No issues detected!`

#### Here's what I wanted or expected to see:

`Warning: glob and single quotes are not recognized as special characters in "${foo-'/bar'*}" (with double quotes)`
